### PR TITLE
Sol Blog Dapp tutorial

### PR DESCRIPTION
**Pull request by kiran-bhalerao**
Thanks for submitting this tutorial for Figment Learn!

Before you hit "Create Pull Request" make sure you have read the [Tutorial Contribution Guidelines](https://www.notion.so/figmentnetworks/Figment-Learn-Contributor-Guidelines-b08691366e144d33a5636d9276a44ec8).
If you do not follow these guidelines, it might take longer for us to review your tutorial and get it published on Learn.

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://www.notion.so/figmentnetworks/Figment-Learn-Contributor-Guidelines-b08691366e144d33a5636d9276a44ec8).

In particular:

- [x] I've followed the [tutorial template](https://www.notion.so/figmentnetworks/Sample-Tutorial-Structure-667ac2aad36f4f94a3ebaca053180b2d) structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run the tutorial to test it and the code works.

### CODE REVIEW

File: network-documentation/solana/tutorials/sol-blog-dapp.md [hunk 1]
```diff
@@ -0,0 +1,831 @@
...
+});
+```
+
+Run again:
+
+```
+anchor test
+```
+
+## Deploying to Devnet
```
Comment (yash-sharma1) [inline network-documentation/solana/tutorials/sol-blog-dapp.md:793]: You created the Solana Blog program and also written down the tests for Blog prgram but you didn't mention about the front end of the Blog dApp, it necessary to have a front-end and connect front end with the deployed Solana prgram. Please add these steps too. It shoudl be a fully functonal dApp with front end. Once you have a front end make sure to include GIF of it in the tutorial.

File: network-documentation/solana/tutorials/sol-blog-dapp.md [hunk 2]
```diff
@@ -0,0 +1,831 @@
...
+anchor build
+```
+
+4. Deploy program
+
+```
+anchor deploy
+```
+
+## Conclusion
```
Comment (yash-sharma1) [inline network-documentation/solana/tutorials/sol-blog-dapp.md:821]: Conclusion should be more detailed. At least 5-6 lines explain what all we have learned in teh tutorial, what should be the next step after this. For ref. please other published tutorials on Learn.

File: network-documentation/solana/tutorials/sol-blog-dapp.md [hunk 3]
```diff
@@ -0,0 +1,831 @@
...
+
+Congratulations on finishing the tutorial. Thank you for taking the time to complete it.
+
+## About Author
+
+This tutorial was created by [Kiran Bhalerao](https://github.com/kiran-bhalerao). If you have any suggestions/questions feel free to contact with author.
```
Comment (yash-sharma1) [inline network-documentation/solana/tutorials/sol-blog-dapp.md:827]: It should be -
This tutorial was created by [Kiran Bhalerao](https://github.com/kiran-bhalerao). For any suggestions/questions you can connect with author on Figment Forum (add link to your Figment Forum profile)

**Comment by yash-sharma1** [comment]
@kiran-bhalerao can you please make the required changes as suggested in the first review of the tutorial? As soon as you'll make these changes we'll give it a final review before merging it :)

File: network-documentation/solana/tutorials/sol-blog-dapp.md [hunk 4]
```diff
@@ -0,0 +1,1327 @@
...
+
+Next, run the test:
+
+```
+anchor test
+```
+
+After running `anchor test` you will see the 1/1 test passing.
+
+Now we complete remaining tests:
```
[the remark this answers is not in the document]
Comment (pyr0gan) [reply network-documentation/solana/tutorials/sol-blog-dapp.md:662]: It is still unclear which file this code belongs in. Please provide the filename where users need to add this code.

File: network-documentation/solana/tutorials/sol-blog-dapp.md [hunk 5]
```diff
@@ -0,0 +1,1327 @@
...
+# Building frontend
+
+Now, We're ready to build out the front end. We will create react app inside the existing app directory.
+
+```text
+cd app
+npx create-react-app .
+```
+
+folder structure of React spp
```
Comment (pyr0gan) [inline network-documentation/solana/tutorials/sol-blog-dapp.md:819]: > The directory structure of a basic React app made with `create-react-app`:

**Review by pyr0gan — CHANGES_REQUESTED** [review]
Hey @kiran-bhalerao,

Some more fixes to make before we can publish, but nothing too difficult. Just have a look over my comments and make the necessary changes 👍

**Comment by kiran-bhalerao** [comment]
@pyr0gan made all the fixes. please review.

**Review by pyr0gan — APPROVED** [review]

**Pull request merged**